src/Orleans/Streams/Internal/StreamSubscriptionHandleImpl.cs

```diff
@@ -121,13 +115,6 @@ public override Task<StreamSubscriptionHandle<T>> ResumeAsync(IAsyncObserver<T>
 
             await NextItem(item, currentToken);
 
-            // check again, in case the expectedToken was changed indiretly via ResumeAsync()
-            if (expectedToken != null)
```

Comment:
It's legacy code left over from before we refactored the StreamSubscriptionHandleImpl.  It was misleading and hindered investigation, so I removed it when I settled on this solution.